Cargo.toml

```diff
@@ -2,7 +2,7 @@
 name = "phonenumber"
 version = "0.3.3+8.13.9"
 edition = "2021"
-rust-version = "1.58.0"
+rust-version = "1.65"  # because of regex 1.7
```

Comment:
Is there a way to remain on 1.62? I don't necessarily know why we mandate `regex = ~1.7`, but I suppose there are a lot of other dependencies that will require 1.62+... 

Reason I'd like to remain compatible with 1.62 is that Sailfish OS will have that as a compiler for the next few months.

If you don't feel like figuring it out, I can try to free up some time to traverse the dependency tree.

As a general remark: there is little reason to bump dependencies (like `either`) in `Cargo.toml` on feature level, if we don't require that feature level. The `Cargo.toml` should indicate the minimum required version of a crate.

Reply:
Hm, then you have an opposite look at Cargo.toml of what I do, I tend to keep dependencies at the latest possible version (given a MSRV) to get all the new bugfixes, performance improvements and features. For now I've downgraded them all to the lowest possible version given the MSRV of 1.62

Reply:
> I tend to keep dependencies at the latest possible version (given a MSRV) to get all the new bugfixes, performance improvements and features.

That is generally the idea, but specifying `regex = "1"` means that by default, the latest in the 1-series will be used. There's no direct need to depend on 1.7 if we don't use 1.7 features.